shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4

```diff
@@ -121,7 +121,223 @@ unionClause
     ;
 
 queryBlock
-    : SELECT duplicateSpecification? projections fromClause? whereClause? groupByClause? havingClause?
+    : withClause? SELECT duplicateSpecification? projections fromClause? whereClause? groupByClause? havingClause?
+    ;
+
+withClause
+    : WITH plsqlDeclarations? ((subqueryFactoringClause | subavFactoringClause) (COMMA_ (subqueryFactoringClause | subavFactoringClause))*)?
+    ;
+
+plsqlDeclarations
+    : (functionDeclaration | procedureDeclaration)+
+    ;
+
+functionDeclaration
+    : functionHeading ((DETERMINISTIC | PIPELINED | PARALLEL_ENABLE | RESULT_CACHE)+)?
+    ;
+
+functionHeading
+    : FUNCTION functionName (LP_ parameterDeclaration (SQ_ parameterDeclaration)* RP_)? RETURN dataType
+    ;
+
+parameterDeclaration
+    : parameterName ((IN? dataType ((COLON_ EQ_ | DEFAULT) expr)?) | (IN? OUT NOCOPY? dataType))?
+    ;
+
+procedureDeclaration
+    : procedureHeading procedureProperties
+    ;
+
+procedureHeading
+    : PROCEDURE procedureName (LP_ parameterDeclaration (SQ_ parameterDeclaration)* RP_)?
+    ;
+
+procedureProperties
+    : (accessibleByClause | defaultCollationClause | invokerRightsClause)*
+    ;
+
+accessibleByClause
+    : ACCESSIBLE BY LP_ accessor (COMMA_ accessor)* RP_
+    ;
+
+accessor
+    : unitKind? unitName
+    ;
+
+unitKind
+    : FUNCTION | PROCEDURE | PACKAGE | TRIGGER | TYPE
+    ;
+
+defaultCollationClause
+    : DEFAULT COLLATION collationOption
+    ;
+
+collationOption
+    : USING_NLS_COMP
+    ;
+
+invokerRightsClause
+    : AUTHID (CURRENT_USER | DEFINER)
+    ;
+
+subqueryFactoringClause
+    : queryName (LP_ alias (COMMA_ alias)* RP_)? AS LP_ selectSubquery RP_ searchClause? cycleClause? 
+    ;
+
+searchClause
+    : SEARCH (DEPTH | BREADTH) FIRST BY (alias (ASC | DESC)? (NULLS FIRST | NULLS LAST)?) (COMMA_ (alias (ASC | DESC)? (NULLS FIRST | NULLS LAST)?))* SET orderingColumn
+    ;
+
+cycleClause
+    : CYCLE alias (COMMA_ alias)* SET alias TO cycleValue DEFAULT noCycleValue
+    ;
+
+subavFactoringClause
+    : subavName ANALYTIC VIEW AS LP_ subavClause RP_
+    ;
+
+subavClause
+    : USING baseAvName hierarchiesClause? filterClauses? addCalcsClause?
+    ;
+
+hierarchiesClause
+    : HIERARCHIES LP_ ((alias DOT_)? alias (COMMA_ (alias DOT_)? alias)*)? RP_
+    ;
+
+filterClauses
+    : FILTER FACT LP_ filterClause (COMMA_ filterClause)* RP_
+    ;
+
+filterClause
+    : (MEASURES | (alias DOT_)? alias) TO expr
+    ;
+
+addCalcsClause
+    : ADD MEASURES LP_ calcMeasClause (COMMA_ calcMeasClause)* RP_
+    ;
+
+calcMeasClause
```

Comment:
Could you put a reference link of the rule `calcMeasClause`?

Reply:
Yes, I'll put it here.

![CalculatedMeasureClause](https://user-images.githubusercontent.com/48581379/120823142-c06a4400-c574-11eb-9354-c8619ba05cdb.png)

Here is the [reference link to `calc_meas_expression`](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/analytic-view-measure-expressions.html#GUID-F8C7ED67-A4EC-479C-975F-12F1F4B8CBA0).